dev/devicelab/lib/framework/utils.dart

```diff
@@ -517,13 +502,20 @@ String extractCloudAuthTokenArg(List<String> rawArgs) {
   return token;
 }
 
-// "An Observatory debugger and profiler on ... is available at: http://127.0.0.1:8100/"
-final RegExp _kObservatoryRegExp = new RegExp(r'An Observatory debugger .* is available at: (\S+:(\d+))');
-
-bool lineContainsServicePort(String line) => line.contains(_kObservatoryRegExp);
-
-int parseServicePort(String line) {
-  final Match match = _kObservatoryRegExp.firstMatch(line);
+/// Tries to extract a port from the string.
+///
+/// The `prefix`, if specified, is a regular expression pattern and must not contain groups.
+///
+/// The `multiLine` flag should be set to true if `line` is actually a buffer of many lines.
+int parseServicePort(String line, {
+  String prefix = 'An Observatory debugger .* is available at: ',
+  bool multiLine = false,
+}) {
+  // e.g. "An Observatory debugger and profiler on ... is available at: http://127.0.0.1:8100/"
+  final RegExp pattern = new RegExp('$prefix(\\S+:(\\d+)/\\S*)\$', multiLine: multiLine);
```

Comment:
This feels pretty fragile, does the Dart team know we depend on this specific line format?
Maybe worth adding a comment where they print this information to emphasize that Flutter depends on it?

Reply:
We don't really depend on it so much as we'll just need to update the pattern when they change.

In reality, it's quite possible that there's people out there who do depend on it. Any time you change anything, you could break people.